src/MatomoTwigExtension.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace MatomoTwigExtension;
+
+use InvalidArgumentException;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+/**
+ * Twig extension for Matomo integration.
+ *
+ * @author Matthieu Napoli <matthieu@mnapoli.fr>
+ */
+class MatomoTwigExtension extends AbstractExtension
+{
+    /**
+     * @var string|null
+     */
+    private $matomoHost;
+
+    /**
+     * @var string|null
+     */
+    private $siteId;
+
+    /**
+     * Useful for disabling Matomo tracking in dev environment.
+     *
+     * @var bool
+     */
+    private $enabled = true;
+
+    public function __construct($matomoHost = null, $siteId = null, $enabled = true)
+    {
+        $this->matomoHost = $matomoHost;
+        $this->siteId = $siteId;
+        $this->enabled = (bool) $enabled;
+    }
+
+    public function getName()
+    {
+        return 'matomo';
+    }
+
+    public function getFunctions()
+    {
+        return [
+            new TwigFunction(
+                'matomo',
+                [$this, 'generateMatomoTrackerCode'],
+                ['is_safe' => ['html']]
+            ),
+        ];
+    }
+
+    /**
+     * @param string|null $matomoHost
+     * @param string|null $siteId
+     *
+     * @return string
+     */
+    public function generateMatomoTrackerCode($matomoHost = null, $siteId = null)
+    {
+        if (!$this->enabled) {
+            return '';
+        }
+
+        $matomoHost = $matomoHost ?: $this->matomoHost;
+        $siteId = $siteId ?: $this->siteId;
+
+        if (null === $matomoHost) {
+            throw new InvalidArgumentException('No Matomo host was configured or given to generate the tracker code');
+        }
+        if (null === $siteId) {
+            throw new InvalidArgumentException('No Matomo site ID was configured or given to generate the tracker code');
+        }
+
+        $matomoHost = rtrim($matomoHost, '/');
+
+        $code = <<<HTML
```

Comment:
You don't need to store the html, just return it and then remove line 98
```suggestion
        return <<<HTML
```